presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java

```diff
@@ -135,6 +137,16 @@ public ListSelectiveStreamReader(
         this.outputType = (ArrayType) outputType.orElse(null);
         this.level = subfieldLevel;
 
+        maxElementLength = subfields.stream()
+                .map(Subfield::getPath)
+                .map(path -> path.get(0))
+                .filter(pathElement -> !(pathElement instanceof Subfield.AllSubscripts))
+                .map(Subfield.LongSubscript.class::cast)
+                .map(Subfield.LongSubscript::getIndex)
+                .mapToInt(Long::intValue)
+                .max()
+                .orElse(-1);
```

Comment:
move -1 to a named constant and use it below

Reply:
added `private static final int ELEMENT_LENGTH_UNBOUNDED = -1;`